project_setup/docker/README.md

```diff
@@ -0,0 +1,264 @@
+can be recreated in other languages through careful configuration of
+`virtualenv`s, `bundler`, etc. In this scenario, we don't need to worry about
+rebuilding images based on requirements (which are effectively dynamically
+included). Further, we can destroy our Docker containers and images with
+little fear as all the libraries are actually stored in the host's file
+system. This approach requires a bit more planning and configuration and
+diverges from the majority of Docker's use cases (meaning documentation is
+harder to find). It also has confusion points around Linux binaries being
+stored on OS X/Windows/etc.
+
+As a third strategy, we can try storing these libraries within a Docker
+volume, meaning they stay with the project between restarts but are not
+(generally) accessible from the host. We'd more commonly use volumes like this
+for database storage, but there's nothing to stop us from storing our runtime
+libraries the same way. This approach has the benefit of being very flexible
+-- our docker-compose configurations aren't closely tied to our specific
+projects, but requires significant planning and configuration.
+
+### One-off commands
+
+Apps generally require more than a single "run-everything" command -- we'll
+want to get to the database console, run tests, migrate data, etc. etc. We
+have several strategies at our disposal.
+
+Before we dive too deep, it's important to first discuss `docker-compose run`.
+Consider
+```sh
+docker-compose run --rm my-service py.test --pdb
+```
+This starts your `my-service` (as defined in your docker-compose manifest),
+including any necessary dependencies, such as databases. It doesn't execute
+`my-service`'s startup command, however; instead it runs `py.test` within the
+`my-service` container. Once that command finished, Docker stop and deletes
+(`--rm`) the running container (any services it depended on will continue to
+run).
+
+As a first approach, we can run these one-off commands directly as described
+above. This works well, but requires engineers keep track of the relevant
+commands and which image they should be ran within. We can soften the burden
+by writing wrapping shell scripts or command aliases.
+
+A second strategy places those commands within the docker-compose manifest as
+pseudo-services, e.g.
+```yml
+services:
+  py.test:
+    image: thing-that-contains-pytest
+    volumes:
+    - $PWD:/apps-dir
+    entrypoint: py.test
+```
+These would be executed via
+```sh
+docker-compose run --rm py.test --pdb
+```
+This approach defines a concise list of the entry points to your software
+suite, but may require additional image rebuilding and can be confusing when
+combined with `docker-compose up`. If taking this approach, be sure to use
+YAML anchors to reduce duplication (search within [this
+example](https://learnxinyminutes.com/docs/yaml/) for more details).
+
+If using Django, Atul has written a [manage.py
```

Comment:
thoughts on linking to Atul's github profile since the world / 18F does not have a uniqueness validation on the name Atul?